src/Features/Core/Portable/EditAndContinue/EditAndContinueDiagnosticUpdateSource.cs

```diff
@@ -48,15 +58,31 @@ public ValueTask<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(Workspace w
         /// We do not track the particular reported diagnostics here since we can just clear all of them at once.
         /// </summary>
         public void ClearDiagnostics()
-            => DiagnosticsCleared?.Invoke(this, EventArgs.Empty);
+        {
+            // No point incrementing if we are going from clean to clean
+            if (_previouslyHadDiagnostics)
+            {
+                _previouslyHadDiagnostics = false;
+                _diagnosticsVersion++;
+            }
+
+            DiagnosticsCleared?.Invoke(this, EventArgs.Empty);
+        }
 
         /// <summary>
         /// Reports given set of project or solution level diagnostics. 
         /// </summary>
-        public void ReportDiagnostics(Workspace workspace, Solution solution, ImmutableArray<DiagnosticData> diagnostics)
+        public void ReportDiagnostics(Workspace workspace, Solution solution, ImmutableArray<DiagnosticData> diagnostics, ImmutableArray<(DocumentId, ImmutableArray<RudeEditDiagnostic> Diagnostics)> rudeEdits)
         {
             RoslynDebug.Assert(solution != null);
 
+            // We don't report rude edits here, we just ensure we consider them in the version number increment
+            if (diagnostics.Any() || rudeEdits.Any(d => d.Diagnostics.Any()))
+            {
+                _previouslyHadDiagnostics = true;
+                _diagnosticsVersion++;
+            }
```

Comment:
not sure i understand the comment, btut he code seems sound.

Reply:
Well I updated the comment.. then I looked and saw you were talking about a completely different comment 😬

Reply:
Updated the right one this time